tests/filecheck/backend/csl/print_csl.mlir

```diff
@@ -39,6 +39,21 @@
     csl.return
   }
 
+  csl.func @casts() {
+    %constI32 = arith.constant 0 : i32
+    %constU16 = arith.constant 0 : ui16
+    %constF32 = arith.constant 0 : f32
+    %castIndex = "arith.index_cast"(%constF32) : (f32) -> index
```

Comment:
`index_cast` can only cast between index and integer data types I'm afraid

Reply:
Not sure why this isn't checked by xDSL

Reply:
I didn't realise that. Fixed

Reply:
PR open #2682 btw :)